capitoli/step4_dati.rst

```diff
@@ -107,47 +123,339 @@ visualizzeranno rispettivamente:
+   :nocanvas:
+   :language: python
+   
    print ("8+4")
    print (8+4)
 
 Visto che non ci sono le virgolette il calcolatore interpreta 8+4 come un’espressione aritmetica e la esegue visualizzandone il risultato.
 
-Proviamo ora a collegare le due istruzioni e scriviamo: print ("8+4  =",  8+4)
+Proviamo ora a collegare le due istruzioni e scriviamo: 
+
+::
+
+	print ("8+4  =",  8+4)
 
 Il calcolatore prima visualizzerà la stringa ”8+4 =“ e poi il risultato dell’operazione 8+4.
 
 | Quale sarà il risultato?  
 | Il messaggio: 8+4 = 12
 
-.. activecode:: esempio_print4
+.. 8+4=12 va blu
+
+.. activecode:: esempio_print5
    :nocanvas:
    :language: python
    
    print ("8+4 =", 8+4)
 
 Facciamo ancora due esempi :
 
-| print  ("3+5=  ", 3+5)
-| Avrà come risultato: 3+5=  8                            
+::
 
-| print  ("4*4= ",  4*4)
-| Avrà come risultato:  4*4=  16
+	print  ("3+5=  ", 3+5)
 
-.. activecode:: esempio_print5
+Avrà come risultato: 3+5=  8                            
+
+::
+
+	print  ("4*4= ",  4*4)
+
+Avrà come risultato:  4*4=  16
+
++------------------+--------------+
+|Sono stringhe     |Sono numeri   |
++==================+==============+
+| "3+5=" "4*4="    | 3+5= 4*4=    |
++------------------+--------------+
+
+.. entrambi i 3+5 vanno rossi, gli altridue blu
+
+Il primo è gia scritto, prova a scrivere il secondo:
+
+.. activecode:: esempio_print6
    :nocanvas:
    :language: python
    
    print ("3+5 =", 3+5)
+
+Nell’esempio successivo cambia solo l’ordine con cui devono essere eseguiti gli ordini, in questo caso prima l’operazione su numeri e poi la stringa:
+
+::
+
+	print (100-10, " = 100 - 10")
+
+Il risultato sarà: 90 = 100-10
+
+.. dopo i due punti scritta verde
+
+.. activecode:: esempio_print7
+   :nocanvas:
+   :language: python
+   
+   print (100-10, " = 100 - 10")
+
+Facciamo due  esempi simili utilizzando le variabili:
+
+::
+
+	scatola2 = 14+8
+	print ("14+8=  ", scatola2)
+
+Il risultato sarà: 14+8=  22
+
+.. dopo i due punti rosso
+
+::
+
+	scatola3 = 5*3
+	print ("5*3=  ", scatola3)
+
+Il risultato sarà: 5*3= 15
+
+.. dopo i due punti rosso
+
+.. activecode:: esempio_scatola2
+   :nocanvas:
+   :language: python
+   
+   scatola2 = 14+8
+   print ("14+8=  ", scatola2)
+
+|
+|
+
+**ATTENZIONE:**
+
+.. attenzione blu
+
+1. La parte a sinistra dell’uguale è una stringa anche se nel messaggio in uscita sembrano tutti numeri. E pertanto 14,  8,  5,  3  non sono numeri ma caratteri.
+
+2. Hai notato che è stata utilizzata una virgola per separare le due parti del messaggio print (“5*3= ”, scatola3) ? Sai perché?
+
+Perché l’'istruzione print() dice a Python: "devi visualizzare le seguenti cose".
+Se vuoi fare visualizzare più oggetti sulla stessa riga devi separarli con una virgola.
+
+Nel tuo caso hai detto a Python: visualizzami la stringa "5*3= " così com'è e subito dopo hai aggiunto: visualizzami il risultato della moltiplicazione tra i numeri 5 e 3.
+
+|
+|
+| Prova a rispondere ai seguenti quesiti:
```

Comment:
Anche qui mi sembra troppo separato, non credi?

Reply:
Invece di mettere tutti questi ritorni a capo metterei una linea orizzontale (`<hr>` in html) che si fa con:
`------------`
Prova e guarda se funziona.
Ad esempio:
testo
<hr>
testo